packages/cli/src/bin/pipcook-plugin.ts

```diff
@@ -46,6 +52,54 @@ async function list(opts: any): Promise<void> {
   }
 }
 
+async function upload(localPluginsPath: string, opts: any): Promise<void> {
+  const spinner = ora();
+  spinner.start(`upload plugin: ${localPluginsPath}`);
+  const params = {
+    pyIndex: opts.tuna ? tunaMirrorURI : undefined
+  };
+  try {
+    const pkg = await fs.readJSON(path.join(localPluginsPath, 'package.json'));
+    if (!pkg?.pipcook) {
+      spinner.fail('invalid plugin package');
+      process.exit(1);
+    }
+  } catch (err) {
+    spinner.fail(`read package.json error: ${err.message}`);
+    process.exit(1);
```

Comment:
How about adding an common function `abort(msg)`?

Reply:
I think we need to extend snipper to exit the process easier.

Reply:
Spinner is our third-party dependency, we cannot extend that.